crates/wasi-http/src/struct.rs

```diff
@@ -8,22 +8,68 @@ pub struct Stream {
     pub data: BytesMut,
 }
 
+impl crate::types::Method {
+    pub fn new(m: &hyper::Method) -> Self {
+        match m {
+            &hyper::Method::GET => Method::Get,
+            &hyper::Method::PUT => Method::Put,
+            &hyper::Method::POST => Method::Post,
+            &hyper::Method::DELETE => Method::Delete,
+            &hyper::Method::OPTIONS => Method::Options,
+            &hyper::Method::HEAD => Method::Head,
+            &hyper::Method::CONNECT => Method::Connect,
+            &hyper::Method::TRACE => Method::Trace,
+            &hyper::Method::PATCH => Method::Patch,
+            _ => panic!("unknown method!"),
+        }
+    }
+}
+
+impl From<&http::uri::Scheme> for crate::types::Scheme {
+    fn from(s: &http::uri::Scheme) -> crate::types::Scheme {
+        match s.as_str() {
+            "http" => crate::types::Scheme::Http,
+            "https" => crate::types::Scheme::Https,
+            _ => panic!("unsupported scheme!"),
+        }
+    }
+}
+
+impl From<crate::types::Method> for u32 {
+    fn from(e: crate::types::Method) -> u32 {
+        match e {
+            Method::Get => 0,
+            Method::Head => 1,
+            Method::Post => 2,
+            Method::Put => 3,
+            Method::Delete => 4,
+            Method::Connect => 5,
+            Method::Options => 6,
+            Method::Trace => 7,
+            Method::Patch => 8,
+            _ => panic!("unknown method"),
```

Comment:
Unfortunately this one is tedious to implement properly, and its probably OK to leave it out for a little bit until we switch over to using the component model runtime
```suggestion
            Method::Other(_) => unimplemented!("Method::Other"),
```